### PR TITLE
feat(rule): enhance formatNumber function for improved precision

### PR DESCRIPTION
**Pull request by AMarcosCastelo**
### 🧾 Summary

_Summarize the suggested changes introduced in this PR._

### 🧩 Context

_Explain **why** this change is being made. If it's part of a larger initiative, describe the broader goal and indicate where this PR fits in._

### 🧱 Scope of this PR

_What is included in this PR? What is intentionally **not** included and will be addressed later?_

### 🧪 How to test

_Provide testing instructions, steps to validate, or environment setup (if needed). Mention feature flags, test accounts, or anything relevant._

### 🧠 Considerations for Review

_List anything that may require special attention during review:_

- Complex logic
- Architectural decisions
- Trade-offs
- Areas to revisit in future PRs

### 🔗 Related Links

_Add links to documentation, Notion pages, ClickUp tasks, Figma designs, or previous/follow-up PRs._

---

- [ ] **I, the PR author, confirm that this PR works as expected and does not break any service. I also confirm that I applied best practices and improved the codebase quality.**

## Summary by CodeRabbit

- New Features
  - Added consistent number formatting to three decimal places with floor rounding, ensuring predictable display for positive, negative, zero, and very small values.
  - Improved handling of edge cases (e.g., values with more than three decimals and trailing decimals) for display and calculations.
- Tests
  - Added comprehensive test coverage validating floor-based truncation to three decimals across diverse scenarios, including very large, very small, negative, and boundary values.

### CODE REVIEW

File: libs/methodologies/bold/rule-processors/mass-id/prevented-emissions/src/prevented-emissions.helpers.ts [hunk 1]
```diff
@@ -91,5 +91,13 @@ export const getGasTypeFromEvent = (
...
 
-export const formatNumber = (number_: number): number =>
-  Math.floor(number_ * 1000) / 1000;
+export const formatNumber = (number_: number): number => {
+  const formatter = new Intl.NumberFormat('en-US', {
+    maximumFractionDigits: 3,
+    minimumFractionDigits: 3,
+    roundingMode: 'floor',
+    useGrouping: false,
+  });
```
Comment (sangalli) [inline libs/methodologies/bold/rule-processors/mass-id/prevented-emissions/src/prevented-emissions.helpers.ts]: We may enhance performance by declaring a static numberFormat and reuse it,

File: libs/methodologies/bold/rule-processors/mass-id/prevented-emissions/src/prevented-emissions.helpers.ts [hunk 2]
```diff
@@ -91,5 +91,13 @@ export const getGasTypeFromEvent = (
   return gasType;
 };
 
-export const formatNumber = (number_: number): number =>
-  Math.floor(number_ * 1000) / 1000;
+export const formatNumber = (number_: number): number => {
+  const formatter = new Intl.NumberFormat('en-US', {
+    maximumFractionDigits: 3,
+    minimumFractionDigits: 3,
```
Comment (sangalli) [inline libs/methodologies/bold/rule-processors/mass-id/prevented-emissions/src/prevented-emissions.helpers.ts]: Do you think we should alway show 3 digits even when they are zero? in my opinion, it would be more user friendly to show 10.31 than 10.310. What do you think?

Comment (AMarcosCastelo) [reply libs/methodologies/bold/rule-processors/mass-id/prevented-emissions/src/prevented-emissions.helpers.ts]: https://github.com/carrot-foundation/methodology-rules/pull/307/commits/f75204c40c354430d1e75e8f24c3729438e45411

**Review by sangalli — APPROVED** [review]
